vm/src/obj/objbyteinner.rs

```diff
@@ -0,0 +1,191 @@
+                        for elem in elements.unwrap(){
+                            let v = objint::to_int(vm, &elem, 10)?;
+                            if let Some(i) = v.to_u8() {
+                                data_bytes.push(i);
+                            } else {
+                                return Err(vm.new_value_error("bytes must be in range(0, 256)".to_string()));
+                                }
+
+                            }
+                        Ok(data_bytes)
+                        }
+                )
+            } else {
+                Ok(vec![])
+            };
+            match value {
+                Ok(val) => Ok(PyByteInner { elements: val }),
+                Err(err) => Err(err),
+            }
+        }
+    }
+
+    pub fn repr(&self) -> PyResult<String> {
+        let mut res = String::with_capacity(self.elements.len());
+        for i in self.elements.iter() {
+            match i {
+                0..=8 => res.push_str(&format!("\\x0{}", i)),
+                9 => res.push_str("\\t"),
+                10 => res.push_str("\\n"),
+                13 => res.push_str("\\r"),
+                32..=126 => res.push(*(i) as char),
+                _ => res.push_str(&format!("\\x{:x}", i)),
+            }
+        }
+        Ok(res)
+    }
+
+    pub fn len(&self) -> usize {
+        self.elements.len()
+    }
+
+    pub fn eq(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        if self.elements == other.elements {
+            Ok(vm.new_bool(true))
+        } else {
+            Ok(vm.new_bool(false))
+        }
+    }
+
+    pub fn ge(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        if self.elements >= other.elements {
+            Ok(vm.new_bool(true))
+        } else {
+            Ok(vm.new_bool(false))
+        }
+    }
+
+    pub fn le(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        if self.elements <= other.elements {
+            Ok(vm.new_bool(true))
+        } else {
+            Ok(vm.new_bool(false))
+        }
+    }
+
+    pub fn gt(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        if self.elements > other.elements {
+            Ok(vm.new_bool(true))
+        } else {
+            Ok(vm.new_bool(false))
+        }
+    }
+
+    pub fn lt(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        if self.elements < other.elements {
+            Ok(vm.new_bool(true))
+        } else {
+            Ok(vm.new_bool(false))
+        }
+    }
+
+    pub fn hash(&self) -> usize {
+        let mut hasher = DefaultHasher::new();
+        self.elements.hash(&mut hasher);
+        hasher.finish() as usize
+    }
+
+    pub fn add(&self, other: &PyByteInner, vm: &VirtualMachine) -> PyResult {
+        let elements: Vec<u8> = self
+            .elements
+            .iter()
+            .chain(other.elements.iter())
+            .cloned()
+            .collect();
+        Ok(vm.ctx.new_bytes(elements))
```

Comment:
This should probably return a new `PyByteInner` and not `bytes`.

Reply:
I finally did it with Vec[u8] instead , easier to call wm.ctx.new_bytes()

Reply:
But thinking to this, maybe it's more consistent to always return a PyByteInner when it's possible. Anyone has an opinion on it ?